Copy with nested pointer to struct causes panic

The following code causes a panic:

```go
package main

import "github.com/jinzhu/copier"

func main() {
	type nested struct {
		A string
	}
	type parentA struct {
		*nested
	}
	type parentB struct {
		*nested
	}
	a := parentA{nested: &nested{A: "a"}}
	b := parentB{}

	copier.Copy(&b, &a)
}

```

stack trace:
```
$ go run copier_panic.go
panic: reflect: indirection through nil pointer to embedded struct

goroutine 1 [running]:
reflect.Value.FieldByIndex(0x4d4fe0, 0xc04206e020, 0x199, 0xc0420500d0, 0x2, 0x2, 0x0, 0x568f40, 0x4c8a00)
        C:/tools/go/src/reflect/value.go:804 +0x276
reflect.Value.FieldByName(0x4d4fe0, 0xc04206e020, 0x199, 0x4b5ea6, 0x1, 0x4c8a00, 0xc042044250, 0x198)
        C:/tools/go/src/reflect/value.go:820 +0x16e
github.com/jinzhu/copier.Copy(0x4c2a60, 0xc04206e020, 0x4c2a20, 0xc04206e018, 0x0, 0x0)
        E:/workspace/golang/src/github.com/jinzhu/copier/copier.go:71 +0x491
main.main()
        E:/workspace/golang/src/playground/copier_panic.go:18 +0xb1
exit status 2
```

The issue originates from this:
https://github.com/jinzhu/copier/blob/db4671f3a9b8df855e993f7c94ec5ef1ffb0a23b/copier.go#L71

Is this something that can be fixed in `copier`? I don't know enough about reflection but if you can guide me, I can make a PR to fix this.

--- a/copier.go
+++ b/copier.go
@@ -194,7 +194,35 @@ func copier(toValue interface{}, fromValue interface{}, opt Option) (err error) 
 
 				if fromField := source.FieldByName(name); fromField.IsValid() && !shouldIgnore(fromField, opt.IgnoreEmpty) {
 					// has field
-					if toField := dest.FieldByName(name); toField.IsValid() {
+
+					// handle anonymous field
+					destFieldNotSet := false
+					if f, ok := dest.Type().FieldByName(name); ok {
+						for _, x := range f.Index {
+							destFieldKind := dest.Field(x).Kind()
+							if destFieldKind != reflect.Ptr {
+								continue
+							}
+
+							if !dest.Field(x).IsNil() {
+								continue
+							}
+
+							if !dest.Field(x).CanSet() {
+								destFieldNotSet = true
+								break
+							}
+
+							newValue := reflect.New(dest.Field(x).Type().Elem())
+							dest.Field(x).Set(newValue)
+						}
+					}
+
+					toField := reflect.Value{}
+					if !destFieldNotSet {
+						toField = dest.FieldByName(name)
+					}
+					if toField.IsValid() {
 						if toField.CanSet() {
 							if !set(toField, fromField, opt.DeepCopy) {
 								if err := copier(toField.Addr().Interface(), fromField.Interface(), opt); err != nil {

--- a/copier_test.go
+++ b/copier_test.go
@@ -818,6 +818,125 @@ func TestSlice(t *testing.T) {
 	})
 }
 
+func TestAnonymousFields(t *testing.T) {
+	t.Run("Should work with unexported ptr fields", func(t *testing.T) {
+		type nested struct {
+			A string
+		}
+		type parentA struct {
+			*nested
+		}
+		type parentB struct {
+			*nested
+		}
+
+		from := parentA{nested: &nested{A: "a"}}
+		to := parentB{}
+
+		err := copier.CopyWithOption(&to, &from, copier.Option{
+			DeepCopy: true,
+		})
+		if err != nil {
+			t.Errorf("Unexpected error: %v", err)
+			return
+		}
+
+		from.nested.A = "b"
+
+		if to.nested != nil {
+			t.Errorf("should be nil")
+		}
+	})
+	t.Run("Should work with unexported fields", func(t *testing.T) {
+		type nested struct {
+			A string
+		}
+		type parentA struct {
+			nested
+		}
+		type parentB struct {
+			nested
+		}
+
+		from := parentA{nested: nested{A: "a"}}
+		to := parentB{}
+
+		err := copier.CopyWithOption(&to, &from, copier.Option{
+			DeepCopy: true,
+		})
+		if err != nil {
+			t.Errorf("Unexpected error: %v", err)
+			return
+		}
+
+		from.nested.A = "b"
+
+		if to.nested.A == from.nested.A {
+			t.Errorf("should be different")
+		}
+	})
+
+	t.Run("Should work with exported ptr fields", func(t *testing.T) {
+		type Nested struct {
+			A string
+		}
+		type parentA struct {
+			*Nested
+		}
+		type parentB struct {
+			*Nested
+		}
+
+		fieldValue := "a"
+		from := parentA{Nested: &Nested{A: fieldValue}}
+		to := parentB{}
+
+		err := copier.CopyWithOption(&to, &from, copier.Option{
+			DeepCopy: true,
+		})
+		if err != nil {
+			t.Errorf("Unexpected error: %v", err)
+			return
+		}
+
+		from.Nested.A = "b"
+
+		if to.Nested.A != fieldValue {
+			t.Errorf("should not change")
+		}
+	})
+
+	t.Run("Should work with exported fields", func(t *testing.T) {
+		type Nested struct {
+			A string
+		}
+		type parentA struct {
+			Nested
+		}
+		type parentB struct {
+			Nested
+		}
+
+		fieldValue := "a"
+		from := parentA{Nested: Nested{A: fieldValue}}
+		to := parentB{}
+
+		err := copier.CopyWithOption(&to, &from, copier.Option{
+			DeepCopy: true,
+		})
+		if err != nil {
+			t.Errorf("Unexpected error: %v", err)
+			return
+		}
+
+		from.Nested.A = "b"
+
+		if to.Nested.A != fieldValue {
+			t.Errorf("should not change")
+		}
+	})
+}
+
 type someStruct struct {
 	IntField  int
 	UIntField uint64

